File: OPENAPI-REST-API/swagger-client/java-vertx/src/main/java/io/swagger/server/api/model/Cpu.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import java.math.BigDecimal;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * A CPU option available for dedicated server ordering.
 */




public class Cpu   {
  private Integer id = null;

  private String shortDesc = null;

  private String longDesc = null;

  private String type = null;

  private String speed = null;

  private String numCores = null;

  private String numCpus = null;

  private String benchmark = null;

  private BigDecimal monthlyPrice = null;

  private String monthlyPriceDisplay = null;

  private String maxRam = null;

  private String minRam = null;

  private String maxLff = null;

  private String maxSff = null;

  private String maxNve = null;

  private String visible = null;

  private String active = null;

  public Cpu id(Integer id) {
    this.id = id;
    return this;
  }

  /**
   * Get id
   * @return id
   **/
    public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public Cpu shortDesc(String shortDesc) {
    this.shortDesc = shortDesc;
    return this;
  }

  /**
   * Get shortDesc
   * @return shortDesc
   **/
    public String getShortDesc() {
    return shortDesc;
  }

  public void setShortDesc(String shortDesc) {
    this.shortDesc = shortDesc;
  }

  public Cpu longDesc(String longDesc) {
    this.longDesc = longDesc;
    return this;
  }

  /**
   * Get longDesc
   * @return longDesc
   **/
    public String getLongDesc() {
    return longDesc;
  }

  public void setLongDesc(String longDesc) {
    this.longDesc = longDesc;
  }

  public Cpu type(String type) {
    this.type = type;
    return this;
  }

  /**
   * Get type
   * @return type
   **/
    public String getType() {
    return type;
  }

  public void setType(String type) {
    this.type = type;
  }

  public Cpu speed(String speed) {
    this.speed = speed;
    return this;
  }

  /**
   * Get speed
   * @return speed
   **/
    public String getSpeed() {
    return speed;
  }

  public void setSpeed(String speed) {
    this.speed = speed;
  }

  public Cpu numCores(String numCores) {
    this.numCores = numCores;
    return this;
  }

  /**
   * Get numCores
   * @return numCores
   **/
    public String getNumCores() {
    return numCores;
  }

  public void setNumCores(String numCores) {
    this.numCores = numCores;
  }

  public Cpu numCpus(String numCpus) {
    this.numCpus = numCpus;
    return this;
  }

  /**
   * Get numCpus
   * @return numCpus
   **/
    public String getNumCpus() {
    return numCpus;
  }

  public void setNumCpus(String numCpus) {
    this.numCpus = numCpus;
  }

  public Cpu benchmark(String benchmark) {
    this.benchmark = benchmark;
    return this;
  }

  /**
   * Get benchmark
   * @return benchmark
   **/
    public String getBenchmark() {
    return benchmark;
  }

  public void setBenchmark(String benchmark) {
    this.benchmark = benchmark;
  }

  public Cpu monthlyPrice(BigDecimal monthlyPrice) {
    this.monthlyPrice = monthlyPrice;
    return this;
  }

  /**
   * Get monthlyPrice
   * @return monthlyPrice
   **/
    public BigDecimal getMonthlyPrice() {
    return monthlyPrice;
  }

  public void setMonthlyPrice(BigDecimal monthlyPrice) {
    this.monthlyPrice = monthlyPrice;
  }

  public Cpu monthlyPriceDisplay(String monthlyPriceDisplay) {
    this.monthlyPriceDisplay = monthlyPriceDisplay;
    return this;
  }

  /**
   * Get monthlyPriceDisplay
   * @return monthlyPriceDisplay
   **/
    public String getMonthlyPriceDisplay() {
    return monthlyPriceDisplay;
  }

  public void setMonthlyPriceDisplay(String monthlyPriceDisplay) {
    this.monthlyPriceDisplay = monthlyPriceDisplay;
  }

  public Cpu maxRam(String maxRam) {
    this.maxRam = maxRam;
    return this;
  }

  /**
   * Get maxRam
   * @return maxRam
   **/
    public String getMaxRam() {
    return maxRam;
  }

  public void setMaxRam(String maxRam) {
    this.maxRam = maxRam;
  }

  public Cpu minRam(String minRam) {
    this.minRam = minRam;
    return this;
  }

  /**
   * Get minRam
   * @return minRam
   **/
    public String getMinRam() {
    return minRam;
  }

  public void setMinRam(String minRam) {
    this.minRam = minRam;
  }

  public Cpu maxLff(String maxLff) {
    this.maxLff = maxLff;
    return this;
  }

  /**
   * Get maxLff
   * @return maxLff
   **/
    public String getMaxLff() {
    return maxLff;
  }

  public void setMaxLff(String maxLff) {
    this.maxLff = maxLff;
  }

  public Cpu maxSff(String maxSff) {
    this.maxSff = maxSff;
    return this;
  }

  /**
   * Get maxSff
   * @return maxSff
   **/
    public String getMaxSff() {
    return maxSff;
  }

  public void setMaxSff(String maxSff) {
    this.maxSff = maxSff;
  }

  public Cpu maxNve(String maxNve) {
    this.maxNve = maxNve;
    return this;
  }

  /**
   * Get maxNve
   * @return maxNve
   **/
    public String getMaxNve() {
    return maxNve;
  }

  public void setMaxNve(String maxNve) {
    this.maxNve = maxNve;
  }

  public Cpu visible(String visible) {
    this.visible = visible;
    return this;
  }

  /**
   * Get visible
   * @return visible
   **/
    public String getVisible() {
    return visible;
  }

  public void setVisible(String visible) {
    this.visible = visible;
  }

  public Cpu active(String active) {
    this.active = active;
    return this;
  }

  /**
   * Get active
   * @return active
   **/
    public String getActive() {
    return active;
  }

  public void setActive(String active) {
    this.active = active;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Cpu cpu = (Cpu) o;
    return Objects.equals(this.id, cpu.id) &&
        Objects.equals(this.shortDesc, cpu.shortDesc) &&
        Objects.equals(this.longDesc, cpu.longDesc) &&
        Objects.equals(this.type, cpu.type) &&
        Objects.equals(this.speed, cpu.speed) &&
        Objects.equals(this.numCores, cpu.numCores) &&
        Objects.equals(this.numCpus, cpu.numCpus) &&
        Objects.equals(this.benchmark, cpu.benchmark) &&
        Objects.equals(this.monthlyPrice, cpu.monthlyPrice) &&
        Objects.equals(this.monthlyPriceDisplay, cpu.monthlyPriceDisplay) &&
        Objects.equals(this.maxRam, cpu.maxRam) &&
        Objects.equals(this.minRam, cpu.minRam) &&
        Objects.equals(this.maxLff, cpu.maxLff) &&
        Objects.equals(this.maxSff, cpu.maxSff) &&
        Objects.equals(this.maxNve, cpu.maxNve) &&
        Objects.equals(this.visible, cpu.visible) &&
        Objects.equals(this.active, cpu.active);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, shortDesc, longDesc, type, speed, numCores, numCpus, benchmark, monthlyPrice, monthlyPriceDisplay, maxRam, minRam, maxLff, maxSff, maxNve, visible, active);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Cpu {\n");
    
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    shortDesc: ").append(toIndentedString(shortDesc)).append("\n");
    sb.append("    longDesc: ").append(toIndentedString(longDesc)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    speed: ").append(toIndentedString(speed)).append("\n");
    sb.append("    numCores: ").append(toIndentedString(numCores)).append("\n");
    sb.append("    numCpus: ").append(toIndentedString(numCpus)).append("\n");
    sb.append("    benchmark: ").append(toIndentedString(benchmark)).append("\n");
    sb.append("    monthlyPrice: ").append(toIndentedString(monthlyPrice)).append("\n");
    sb.append("    monthlyPriceDisplay: ").append(toIndentedString(monthlyPriceDisplay)).append("\n");
    sb.append("    maxRam: ").append(toIndentedString(maxRam)).append("\n");
    sb.append("    minRam: ").append(toIndentedString(minRam)).append("\n");
    sb.append("    maxLff: ").append(toIndentedString(maxLff)).append("\n");
    sb.append("    maxSff: ").append(toIndentedString(maxSff)).append("\n");
    sb.append("    maxNve: ").append(toIndentedString(maxNve)).append("\n");
    sb.append("    visible: ").append(toIndentedString(visible)).append("\n");
    sb.append("    active: ").append(toIndentedString(active)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/Cpu.cpp


#include "Cpu.h"

using namespace Tiny;

Cpu::Cpu()
{
	id = int(0);
	short_desc = std::string();
	long_desc = std::string();
	type = std::string();
	speed = std::string();
	num_cores = std::string();
	num_cpus = std::string();
	benchmark = std::string();
	monthly_price = float(0);
	monthly_price_display = std::string();
	max_ram = std::string();
	min_ram = std::string();
	max_lff = std::string();
	max_sff = std::string();
	max_nve = std::string();
	visible = std::string();
	active = std::string();
}

Cpu::Cpu(std::string jsonString)
{
	this->fromJson(jsonString);
}

Cpu::~Cpu()
{

}

void
Cpu::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *idKey = "id";

    if(object.has_key(idKey))
    {
        bourne::json value = object[idKey];



        jsonToValue(&id, value, "int");


    }

    const char *short_descKey = "short_desc";

    if(object.has_key(short_descKey))
    {
        bourne::json value = object[short_descKey];



        jsonToValue(&short_desc, value, "std::string");


    }

    const char *long_descKey = "long_desc";

    if(object.has_key(long_descKey))
    {
        bourne::json value = object[long_descKey];



        jsonToValue(&long_desc, value, "std::string");


    }

    const char *typeKey = "type";

    if(object.has_key(typeKey))
    {
        bourne::json value = object[typeKey];



        jsonToValue(&type, value, "std::string");


    }

    const char *speedKey = "speed";

    if(object.has_key(speedKey))
    {
        bourne::json value = object[speedKey];



        jsonToValue(&speed, value, "std::string");


    }

    const char *num_coresKey = "num_cores";

    if(object.has_key(num_coresKey))
    {
        bourne::json value = object[num_coresKey];



        jsonToValue(&num_cores, value, "std::string");


    }

    const char *num_cpusKey = "num_cpus";

    if(object.has_key(num_cpusKey))
    {
        bourne::json value = object[num_cpusKey];



        jsonToValue(&num_cpus, value, "std::string");


    }

    const char *benchmarkKey = "benchmark";

    if(object.has_key(benchmarkKey))
    {
        bourne::json value = object[benchmarkKey];



        jsonToValue(&benchmark, value, "std::string");


    }

    const char *monthly_priceKey = "monthly_price";

    if(object.has_key(monthly_priceKey))
    {
        bourne::json value = object[monthly_priceKey];



        jsonToValue(&monthly_price, value, "long");


    }

    const char *monthly_price_displayKey = "monthly_price_display";

    if(object.has_key(monthly_price_displayKey))
    {
        bourne::json value = object[monthly_price_displayKey];



        jsonToValue(&monthly_price_display, value, "std::string");


    }

    const char *max_ramKey = "max_ram";

    if(object.has_key(max_ramKey))
    {
        bourne::json value = object[max_ramKey];



        jsonToValue(&max_ram, value, "std::string");


    }

    const char *min_ramKey = "min_ram";

    if(object.has_key(min_ramKey))
    {
        bourne::json value = object[min_ramKey];



        jsonToValue(&min_ram, value, "std::string");


    }

    const char *max_lffKey = "max_lff";

    if(object.has_key(max_lffKey))
    {
        bourne::json value = object[max_lffKey];



        jsonToValue(&max_lff, value, "std::string");


    }

    const char *max_sffKey = "max_sff";

    if(object.has_key(max_sffKey))
    {
        bourne::json value = object[max_sffKey];



        jsonToValue(&max_sff, value, "std::string");


    }

    const char *max_nveKey = "max_nve";

    if(object.has_key(max_nveKey))
    {
        bourne::json value = object[max_nveKey];



        jsonToValue(&max_nve, value, "std::string");


    }

    const char *visibleKey = "visible";

    if(object.has_key(visibleKey))
    {
        bourne::json value = object[visibleKey];



        jsonToValue(&visible, value, "std::string");


    }

    const char *activeKey = "active";

    if(object.has_key(activeKey))
    {
        bourne::json value = object[activeKey];



        jsonToValue(&active, value, "std::string");


    }


}

bourne::json
Cpu::toJson()
{
    bourne::json object = bourne::json::object();





    object["id"] = getId();






    object["short_desc"] = getShortDesc();






    object["long_desc"] = getLongDesc();






    object["type"] = getType();






    object["speed"] = getSpeed();






    object["num_cores"] = getNumCores();






    object["num_cpus"] = getNumCpus();






    object["benchmark"] = getBenchmark();






    object["monthly_price"] = getMonthlyPrice();






    object["monthly_price_display"] = getMonthlyPriceDisplay();






    object["max_ram"] = getMaxRam();






    object["min_ram"] = getMinRam();






    object["max_lff"] = getMaxLff();






    object["max_sff"] = getMaxSff();






    object["max_nve"] = getMaxNve();






    object["visible"] = getVisible();






    object["active"] = getActive();



    return object;

}

int
Cpu::getId()
{
	return id;
}

void
Cpu::setId(int id)
{
	this->id = id;
}

std::string
Cpu::getShortDesc()
{
	return short_desc;
}

void
Cpu::setShortDesc(std::string short_desc)
{
	this->short_desc = short_desc;
}

std::string
Cpu::getLongDesc()
{
	return long_desc;
}

void
Cpu::setLongDesc(std::string long_desc)
{
	this->long_desc = long_desc;
}

std::string
Cpu::getType()
{
	return type;
}

void
Cpu::setType(std::string type)
{
	this->type = type;
}

std::string
Cpu::getSpeed()
{
	return speed;
}

void
Cpu::setSpeed(std::string speed)
{
	this->speed = speed;
}

std::string
Cpu::getNumCores()
{
	return num_cores;
}

void
Cpu::setNumCores(std::string num_cores)
{
	this->num_cores = num_cores;
}

std::string
Cpu::getNumCpus()
{
	return num_cpus;
}

void
Cpu::setNumCpus(std::string num_cpus)
{
	this->num_cpus = num_cpus;
}

std::string
Cpu::getBenchmark()
{
	return benchmark;
}

void
Cpu::setBenchmark(std::string benchmark)
{
	this->benchmark = benchmark;
}

long
Cpu::getMonthlyPrice()
{
	return monthly_price;
}

void
Cpu::setMonthlyPrice(long monthly_price)
{
	this->monthly_price = monthly_price;
}

std::string
Cpu::getMonthlyPriceDisplay()
{
	return monthly_price_display;
}

void
Cpu::setMonthlyPriceDisplay(std::string monthly_price_display)
{
	this->monthly_price_display = monthly_price_display;
}

std::string
Cpu::getMaxRam()
{
	return max_ram;
}

void
Cpu::setMaxRam(std::string max_ram)
{
	this->max_ram = max_ram;
}

std::string
Cpu::getMinRam()
{
	return min_ram;
}

void
Cpu::setMinRam(std::string min_ram)
{
	this->min_ram = min_ram;
}

std::string
Cpu::getMaxLff()
{
	return max_lff;
}

void
Cpu::setMaxLff(std::string max_lff)
{
	this->max_lff = max_lff;
}

std::string
Cpu::getMaxSff()
{
	return max_sff;
}

void
Cpu::setMaxSff(std::string max_sff)
{
	this->max_sff = max_sff;
}

std::string
Cpu::getMaxNve()
{
	return max_nve;
}

void
Cpu::setMaxNve(std::string max_nve)
{
	this->max_nve = max_nve;
}

std::string
Cpu::getVisible()
{
	return visible;
}

void
Cpu::setVisible(std::string visible)
{
	this->visible = visible;
}

std::string
Cpu::getActive()
{
	return active;
}

void
Cpu::setActive(std::string active)
{
	this->active = active;
}



